Framework/API/src/Sample.cpp

```diff
@@ -363,7 +363,11 @@ int Sample::loadNexus(::NeXus::File *file, const std::string &group) {
     }
     Kernel::Material material;
     material.loadNexus(file, "material");
-    m_shape->setMaterial(material);
+    // CSGObject expected, if so, set its material
+    if (auto csgObj =
```

Comment:
`ShapeFactory` should already be creating a `CSGObject` for you. All you have to do is assign that to a temporary, set the material (no dynamic cast required), and then set `m_shape` from the temporary.